packages/core/src/api/toDecimal.ts

```diff
@@ -32,9 +41,15 @@ export function toDecimal<TAmount, TOutput>(calculator: Calculator<TAmount>) {
     assert(isDecimal, NON_DECIMAL_CURRENCY_MESSAGE);
 
     const units = toUnitsFn(dineroObject);
+    const isNegative = units.some((unit) => lessThanFn(signFn(unit), zero));
 
     const getDecimalFn = getDecimal(calculator, dineroObject.formatter);
-    const value = getDecimalFn(units, scale);
+    const decimalValue = getDecimalFn(units, scale);
```

Comment:
I would probably perform the two checks (whether it leads with zero and is negative) inside `getDecimal`—I expect this function to return the correct value. wdyt?